### PR TITLE
Db/user can manage secrets

### PR DESCRIPTION
**Pull request by davidbloss**
Add secrets management as specified in [issue 38](https://github.com/OpsLevel/team-platform/issues/36)

Users will be able to:
- create
- list
- get
- update
- delete
`secrets` 🤫 

I've also placed a good chunk of the test data in template files in preparation for future work to simplify testing.

### CODE REVIEW

File: secrets_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	// Act
+	secretsVaultsSecretConnection, err := client.ListSecretsVaultsSecret(nil)
+	secretNode := secretsVaultsSecretConnection.Nodes
+	// Assert
+	autopilot.Ok(t, err)
+	autopilot.Equals(t, 3, secretsVaultsSecretConnection.TotalCount)
+	autopilot.Equals(t, "example_2", secretNode[1].Alias)
+	autopilot.Equals(t, secretNode[1].Alias, secretNode[1].Owner.Alias)
+}
+
```
Comment (rocktavious) [inline secrets_test.go:60]: missing a test for the GET function.

File: secrets_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  "query": "mutation SecretsVaultsSecretCreate($alias:String!$input:SecretInput!){secretsVaultsSecretCreate(alias: $alias, input: $input){secret{alias,id,owner{alias,id},timestamps{createdAt,updatedAt}},errors{message,path}}}",
+  "variables": {{ template "secret_create_vars" }}
+  }`
+	response := `{{ template "secret_create_response" }}`
+	id2 := opslevel.NewID("Z2lkOi8vOTg3NjU0MzIxMTIzNDU2Nzg5")
+	client := ABetterTestClient(t, "secrets/create", request, response)
+	fmt.Println(client)
+	// Act
+	secretInput := opslevel.SecretInput{
+		Owner: opslevel.IdentifierInput{Id: *id2},
```
Comment (rocktavious) [inline secrets_test.go:23]: Could also use
```suggestion
		Owner: opslevel.NewIdentifier("Z2lkOi8vOTg3NjU0MzIxMTIzNDU2Nzg5"),
```
but i see why you didn.  Just letting you know we have helpers for safely creating IdentifierInput too

**Review by rocktavious — APPROVED** [review]
Looks good - not sure if we have time for the missing test

File: secrets_test.go (hunk 1 above)
Comment (davidbloss) [reply secrets_test.go:75]: Added the test 👍

**Review by rocktavious — APPROVED** [review]

**Pull request merged**